docs/release-notes/RELEASE-NOTES-4.2.0.md

```diff
@@ -0,0 +1,112 @@
+# About this release
+
+[This][this-release] is a maintenance release.
+It includes the following changes:
```

Comment:
@tolbrino Any updates required to user doc for Windows build fix in https://github.com/aeternity/aeternity/pull/2625 ?